modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C

```diff
@@ -200,31 +287,49 @@ PolycrystalUserObjectBase::finalize()
 void
 PolycrystalUserObjectBase::mergeSets()
 {
-  /**
-   * With initial conditions we know the grain IDs of every grain (even partial grains). We can use
-   * this information to put all mergeable features adjacent to one and other in the list so that
-   * merging is simply O(n).
-   */
-  _partial_feature_sets[0].sort();
-
-  auto it1 = _partial_feature_sets[0].begin();
-  auto it_end = _partial_feature_sets[0].end();
-  while (it1 != it_end)
+  // When working with _distribute_merge_work all of the maps will be empty except for one
+  for (MooseIndex(_maps_size) map_num = 0; map_num < _partial_feature_sets.size(); ++map_num)
```

Comment:
```suggestion
  for (const auto map_num: index_range(_partial_feature_sets))
```

... just a suggestion